File: hackerearth-vignan/III-SEM-2022-26/challenge_13/Q6_queueOperations.cpp
#include <iostream>
#include <queue>
using namespace std;

int main() {
    int m;
    cin >> m;
    queue<int> q;
    for (int i = 0; i < m; ++i) {
        int opration;
        cin >> opration;
        if (opration == 1) {
            int val;
            cin >> val;
            q.push(val);
        } else if (opration == 2) {
            if (!q.empty()) {
                q.pop();
            } else {
                cout << -1 << endl;
            }
        } else if (opration == 3) {
            if (!q.empty()) {
                cout << q.front() << endl;
            } else {
                cout << -1 << endl;
            }
        }
    }

    return 0;
}

File: hackerearth-vignan/III-SEM-2022-26/challenge_13/Q6_queueOperations.py
from queue import Queue

m = int(input())
q = Queue()

for _ in range(m):
    operation = int(input())
    if operation == 1:
        val = int(input())
        q.put(val)
    elif operation == 2:
        if not q.empty():
            q.get()
        else:
            print(-1)
    elif operation == 3:
        if not q.empty():
            print(q.queue[0])
        else:
            print(-1)
